plugins/GranularPitchShifter/GranularPitchShifterControlDialog.cpp

```diff
@@ -0,0 +1,158 @@
+		newKnob->setHintText(hintText, unit);
+		return newKnob;
+	};
+
+	makeKnob(KnobType::Bright26, 20, 81, tr("Grain Size:"), " Hz", &controls->m_sizeModel);
+	makeKnob(KnobType::Bright26, 110, 10, tr("Spray:"), " seconds", &controls->m_sprayModel);
+	makeKnob(KnobType::Bright26, 148, 10, tr("Jitter:"), " octaves", &controls->m_jitterModel);
+	makeKnob(KnobType::Bright26, 190, 10, tr("Twitch:"), " octaves", &controls->m_twitchModel);
+	makeKnob(KnobType::Bright26, 184, 63, tr("Spray Stereo Spread:"), "", &controls->m_spraySpreadModel);
+	makeKnob(KnobType::Bright26, 131, 116, tr("Grain Shape:"), "", &controls->m_shapeModel);
+	makeKnob(KnobType::Bright26, 184, 116, tr("Fade Length:"), "", &controls->m_fadeLengthModel);
+	makeKnob(KnobType::Bright26, 248, 48, tr("Feedback:"), "", &controls->m_feedbackModel);
+	makeKnob(KnobType::Bright26, 248, 95, tr("Minimum Allowed Latency:"), " seconds", &controls->m_minLatencyModel);
+	makeKnob(KnobType::Small17, 67, 162, tr("Density:"), "x", &controls->m_densityModel);
+	makeKnob(KnobType::Small17, 9, 162, tr("Glide:"), " seconds", &controls->m_glideModel);
+	
+	LcdFloatSpinBox* pitchBox = new LcdFloatSpinBox(3, 2, "11green", tr("Pitch"), this);
+	pitchBox->move(16, 44);
+	pitchBox->setModel(&controls->m_pitchModel);
+	pitchBox->setToolTip(tr("Pitch"));
+	pitchBox->setSeamless(true, true);
+	
+	LcdFloatSpinBox* pitchSpreadBox = new LcdFloatSpinBox(3, 2, "11green", tr("Pitch Stereo Spread"), this);
+	pitchSpreadBox->move(129, 69);
+	pitchSpreadBox->setModel(&controls->m_pitchSpreadModel);
+	pitchSpreadBox->setToolTip(tr("Pitch Stereo Spread"));
+	pitchSpreadBox->setSeamless(true, true);
+	
+	QPushButton button("Show Help", this);
+	connect(&button, &QPushButton::clicked, this, &GranularPitchShifterControlDialog::showHelpWindow);
+	
+	PixmapButton* m_helpBtn = new PixmapButton(this, nullptr);
+	m_helpBtn->move(272, 164);
+	m_helpBtn->setActiveGraphic(PLUGIN_NAME::getIconPixmap("help_active"));
+	m_helpBtn->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("help_inactive"));
+	m_helpBtn->setToolTip(tr("Open help window"));
+	connect(m_helpBtn, SIGNAL(clicked()), this, SLOT(showHelpWindow()));
+	
+	PixmapButton * prefilterButton = new PixmapButton(this, tr("Prefilter"));
+	prefilterButton->move(8, 139);
+	prefilterButton->setActiveGraphic(PLUGIN_NAME::getIconPixmap("prefilter_active"));
+	prefilterButton->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("prefilter_inactive"));
+	prefilterButton->setCheckable(true);
+	prefilterButton->setModel(&controls->m_prefilterModel);
+	prefilterButton->setToolTip(tr("Prefilter"));
+	
+	ComboBox* rangeBox = new ComboBox(this);
+	rangeBox->setGeometry(181, 160, 83, 22);
+	rangeBox->setModel(&controls->m_rangeModel);
+	controls->updateRange();
+}
+
+void GranularPitchShifterControlDialog::showHelpWindow() {
+	GranularPitchShifterHelpView::getInstance()->close();
+	GranularPitchShifterHelpView::getInstance()->show();
+}
+
+
+QString GranularPitchShifterHelpView::s_helpText=
```

Comment:
I believe it would help if this entire help document was it's own text file and we can read the contents at runtime if the user clicks the help button.

Reply:
Reading those contents at runtime would require shipping that separate text file along with every LMMS copy (and then handling file paths and directory settings and all that), which really isn't necessary.  This help page is handled in the same way as X-Pressive currently.

Reply:
Can't you put that help file in along with the sources at this same folder. You won't have to deal with paths too.

Reply:
I don't really see what the advantage in doing this would be.

Reply:
Just shifting lines to a seperate text file, if this doesn't please you, feel free to keep it as is.